pkg/resources/kafka/pod.go

```diff
@@ -372,39 +373,84 @@ func generateDataVolumeAndVolumeMount(pvcs []corev1.PersistentVolumeClaim) (volu
 	return
 }
 
-func generateVolumesForSSL(clusterName string) []corev1.Volume {
-	return []corev1.Volume{
-		{
-			Name: serverKeystoreVolume,
-			VolumeSource: corev1.VolumeSource{
-				Secret: &corev1.SecretVolumeSource{
-					SecretName:  fmt.Sprintf(pkicommon.BrokerServerCertTemplate, clusterName),
-					DefaultMode: util.Int32Pointer(0644),
-				},
+func generateVolumeForListenersCertsFromCommonSpec(commonSpec v1beta1.CommonListenerSpec, clusterName string) corev1.Volume {
+	// Use default one if custom has not specified
+	secretName := fmt.Sprintf(pkicommon.BrokerControllerTemplate, clusterName)
```

Comment:
Using BrokerControllerTemplate here is wrong. SInce this is the server side we should use the BrokerServerCertTemplate.

Reply:
nice catch